lib/matplotlib/text.py

```diff
@@ -642,10 +642,13 @@ def _get_rendered_text_width(self, text):
         """
         Return the width of a given text string, in pixels.
         """
+        # Determine if text contains mathtext
+        has_math = '$' in text
+
         w, h, d = self._renderer.get_text_width_height_descent(
             text,
             self.get_fontproperties(),
-            False)
+            has_math)
```

Comment:
Determining whether a text contains math is more complicated than "contains $". We have a helper for that:
```suggestion
            cbook.is_math_text(text))
```

Reply:
I shoulda known, thanks for the tip